### PR TITLE
Add HDFS searchable snapshot integration

### PR DESCRIPTION
**Pull request by jbaiera**
Adds a bounded read implementation on the HDFS blob store as well as integration tests to the searchable snapshot project that ensures functionality on both kerberos and simple authentication HDFS.

### CODE REVIEW

**Comment by jbaiera** [comment]
Currently debugging an intermittent SocketPermission exception but the code should be in a place where it's ok for review.

**Comment by jbaiera** [comment]
@elasticmachine update branch

**Comment by jbaiera** [comment]
Update: I've been repeating the tests that were failing intermittently with SocketPermissions through the afternoon. After pushing 6d86e22 they have not resurfaced. We should be good on that front now!

File: plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
     static {
         // We can do FS ops with only a few elevated permissions:
         SIMPLE_AUTH_PERMISSIONS = new Permission[]{
-            new SocketPermission("*", "connect"),
+            new SocketPermission("*", "connect,resolve"),
```
Comment (original-brownbear) [inline plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java:53]: Why do we need `resolve` now, it seems the only new thing we do is a `.seek` call?

**Review by original-brownbear — COMMENTED** [review]
Looks good @jbaiera just one question I think from a quick read through this.

**Review by jasontedor — APPROVED** [review]
Other than the `resolve` permission question that @original-brownbear has, LGTM.

File: plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java (hunk 1 above)
Comment (jbaiera) [reply plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java:53]: likely not. I added this to try to appease the permission errors. Given the problem was something else, it's unlikely that we'll need this.

**Review by original-brownbear — APPROVED** [review]
LGTM (sorry for the delay, somehow missed clicking submit here)

**Pull request merged**